src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -36,6 +36,108 @@
 */
 public with sharing class BGE_BatchGiftEntry_UTIL {
 
+    /*******************************************************************************************************
+    * @description returns a list of DataImport__c fields the Batch Gift Entry UI needs in SOQL
+    * @param batchId the batch for which to get all fields for soql
+    * @return List<String> list of DataImport__c field api names
+    */
+    public static List<String> getDataImportFields(Id batchId) {
+        return getDataImportFields(batchId, true);
+    }
+
+    /*******************************************************************************************************
+    * @description returns a list of DataImport__c fields the Batch Gift Entry UI needs in SOQL
+    * @param batchId the batch for which to get all fields for soql
+    * @param includeRelationshipFields whether to include relationship fields in the returned list
+    * @return List<String> list of DataImport__c field api names
+    */
+    public static List<String> getDataImportFields(Id batchId, Boolean includeRelationshipFields) {
+
+        List<String> fields = getCoreDataImportFields(includeRelationshipFields);
+        fields.addAll(getActiveFieldNamesFromBatch(batchId));
+
+        return fields;
+    }
+
+    /*******************************************************************************************************
+    * @description Returns the subset of DataImport__c fields that are part of every batch
+    * @param includeRelationshipFields whether to include relationship fields in the returned list
+    * @return list of DataImport__c field api names
+    */
+    private static List<String> getCoreDataImportFields(Boolean includeRelationshipFields) {
+        List<String> fields = new List<String> {
+            'Id',
```

Comment:
Is there any reason that Donation_Amount__c, Donation_Date__c, and NPSP_Data_Import_Batch__c aren't included in this list?

Reply:
Yeah, these get added in by the "getActiveFieldNamesFromBatch" method as these are always required and are also considered "selected" fields in the BGE.

Reply:
nm let's leave this alone